itest/lnd_sweep_test.go

```diff
@@ -857,3 +860,319 @@ func createSimpleNetwork(ht *lntest.HarnessTest, nodeCfg []string,
+		require.Equal(ht, op.OutputIndex,
+			pendingSweep.Outpoint.OutputIndex)
+		require.Equal(ht, walletrpc.WitnessType_TAPROOT_PUB_KEY_SPEND,
+			pendingSweep.WitnessType)
+		require.EqualValuesf(ht, value, pendingSweep.AmountSat,
+			"amount not matched: want=%d, got=%d", value,
+			pendingSweep.AmountSat)
+		require.True(ht, pendingSweep.Immediate)
+
+		require.Equal(ht, broadcastAttempts,
+			pendingSweep.BroadcastAttempts)
+		require.EqualValuesf(ht, budget, pendingSweep.Budget,
+			"budget not matched: want=%d, got=%d", budget,
+			pendingSweep.Budget)
+
+		// Since the request doesn't specify a deadline, we expect the
+		// existing deadline to be used.
+		require.Equalf(ht, deadline, pendingSweep.DeadlineHeight,
+			"deadline height not matched: want=%d, got=%d",
+			deadline, pendingSweep.DeadlineHeight)
+
+		// Since the request specifies a starting fee rate, we expect
+		// that to be used as the starting fee rate.
+		require.Equalf(ht, startingFeeRate,
+			pendingSweep.RequestedSatPerVbyte, "requested "+
+				"starting fee rate not matched: want=%d, "+
+				"got=%d", startingFeeRate,
+			pendingSweep.RequestedSatPerVbyte)
+
+		// We expect to see Alice's original tx and her CPFP tx in the
+		// mempool.
+		txns := ht.Miner.GetNumTxsFromMempool(2)
+
+		// Find the sweeping tx - assume it's the first item, if it has
+		// the same txid as the parent tx, use the second item.
+		sweepTx := txns[0]
+		if sweepTx.TxHash() == tx.TxHash() {
+			sweepTx = txns[1]
+		}
+
+		return sweepTx
+	}
+
+	// assertFeeRateEqual is a helper closure that asserts the fee rate of
+	// the pending sweep tx is equal to the expected fee rate.
+	assertFeeRateEqual := func(expected uint64) {
+		err := wait.NoError(func() error {
+			// Alice should still have one pending sweep.
+			pendingSweep := ht.AssertNumPendingSweeps(alice, 1)[0]
+
+			if pendingSweep.SatPerVbyte == expected {
+				return nil
+			}
+
+			return fmt.Errorf("expected current fee rate %d, got "+
+				"%d", expected, pendingSweep.SatPerVbyte)
+
+		}, wait.DefaultTimeout)
+		require.NoError(ht, err, "fee rate not updated")
+	}
+
+	// assertFeeRateGreater is a helper closure that asserts the fee rate
+	// of the pending sweep tx is greater than the expected fee rate.
+	assertFeeRateGreater := func(expected uint64) {
+		err := wait.NoError(func() error {
+			// Alice should still have one pending sweep.
+			pendingSweep := ht.AssertNumPendingSweeps(alice, 1)[0]
+
+			if pendingSweep.SatPerVbyte > expected {
+				return nil
+			}
+
+			return fmt.Errorf("expected current fee rate greater "+
+				"than %d, got %d", expected,
+				pendingSweep.SatPerVbyte)
+
+		}, wait.DefaultTimeout)
+		require.NoError(ht, err, "fee rate not updated")
+	}
+
+	// First bump request - we'll specify nothing except `Immediate` to let
+	// the sweeper handles the fee, and we expect a fee func that has,
+	// - starting fee rate: 1 sat/vbyte (min relay fee rate).
+	// - deadline: 1008 (default deadline).
```

Comment:
it's up to the caller to decide - if they choose to set conf target to 0, it will use the default value. Until we can use that `optional` flag in grpc, we just assume 0 means not set.

Reply:
ok so the user can actually tweak the sweeper in whatever way he wants, cool.